src/components/forms/DataLocationList.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {useCallback} from 'react';
+import {DataLocation, DataLocationInfo} from './DataLocation';
+import {cloneDeep} from 'lodash';
+
+export type DataLocationsProps = {
+    locations: DataLocationInfo[],
+    onChange: (p: { key: string; value: DataLocationInfo[] }) => void
+}
+
+export const DataLocationList = (props: DataLocationsProps) => {
+    const {locations, onChange} = props;
+    const onChangeLocation = useCallback(({idx, location}: { idx: number, location: DataLocationInfo }) => {
+        locations[idx] = location;
+        onChange({key: 'locations', value: cloneDeep(locations)});
+    }, [locations, onChange]);
+
+    const onAddLocation = useCallback(() => {
+        const emptyLocation = {
+            cloudProvider: null,
```

Comment:
If you use `undefined` rather than `null`, this object can be `{}`?

Reply:
Unfortunately, the FormField code wants null.  I actually went down your route first but didn't get the desired behavior out of the FormFileds (input masks don't show up).